scene/3d/collision_object_3d.cpp

```diff
@@ -207,26 +223,40 @@ void CollisionObject3D::_clear_debug_shapes() {
 		ShapeData::ShapeBase *shapes = shapedata.shapes.ptrw();
 		for (int i = 0; i < shapedata.shapes.size(); i++) {
 			ShapeData::ShapeBase &s = shapes[i];
-			if (s.debug_shape) {
-				s.debug_shape->queue_delete();
-				s.debug_shape = nullptr;
-				--debug_shape_count;
+			if (s.debug_shape.is_valid()) {
+				RS::get_singleton()->free(s.debug_shape);
+				s.debug_shape = RID();
+				while (s.shape.is_valid() && !s.shape->is_connected("changed", callable_mp(this, &CollisionObject3D::_update_shape_data))) {
```

Comment:
Shouldn't this be `s.shape->is_connected` instead of `!s.shape->is_connected`? And also why a while loop?
(Same thing in `shape_owner_remove_shape`)

Reply:
It should, and I removed those while (see [this comment](https://github.com/godotengine/godot/pull/48175#discussion_r623753937))